### PR TITLE
Update ice climo, fix option for non-fractional grid, and add cloud fraction for Thompson

### LINKED ISSUE #886: workflow updates: new ice climatology, cplmode=nems_orig for non-fractional grid, add cldfra 3D cloud in all diag_table
A new ICE climatology based IMS and NIC data has been developed to  replace the CFSR ice climatology used by UFS applications.  This new IMS-NIC climatology has a much higher resolution than CFSR and is based on more recent observations. ( Credit: Mike Barlage, Shan Sun,  and many others participated in the [development ](https://docs.google.com/presentation/d/1rf8Nrghtfi-3k0AThRR9rnyVrBzSJoqlFgKqMLJJeJ0/edit#slide=id.gf34d33e052_0_5)and [discussion ](https://docs.google.com/document/d/1PzNtcTwkJyzWAuGrQj4B6g41OcdgZf0fLbLK0OV-000/edit)of this new dataset).  This new dataset has been tested in both coupled and uncoupled UFS experiments. 

1) please copy /scratch1/NCEPDEV/stmp2/Michael.Barlage/data/ice_climatology/IMS-NIC.blended.ice.monthly.clim.grb from Hera to ./fix_NEW/fix_am on all supported platforms.

2) Please update ./ush/forecast_postdet.sh in global workflow
    replace
    FNAISC=${FNAISC:-"$FIX_AM/CFSR.SEAICE.1982.2012.monthly.clim.grb"}
   with 
   FNAISC=${FNAISC:-"$FIX_AM/IMS-NIC.blended.ice.monthly.clim.grb"}

3) make the same change in ./ush/global_cycle.sh

@GeorgeGayno-NOAA  George, please check if you need to make corresponding updates in ufs_util.

### PR DESCRIPTION
**Pull request by yangfanglin**
Updates the ice climatology from CFSR to IMS_NIC blended.

Fixes issue in the diag table where the cloud fraction entry for Thompson was not included.

Adds new coupled mode setting for when fractional grid is not used.

Fixes #886 

**Description**

**Type of change**

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**How Has This Been Tested?**

  
**Checklist**

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes need updates to the documentation. I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] New and existing tests pass with my changes
- [ ] Any dependent changes have been merged and published

### CODE REVIEW

File: parm/parm_fv3diag/diag_table [hunk 1]
```diff
@@ -22,6 +22,7 @@
 "gfs_dyn",     "ps",          "pressfc",      "fv3_history",    "all",  .false.,  "none",  2
 "gfs_dyn",     "hs",          "hgtsfc",       "fv3_history",    "all",  .false.,  "none",  2
 
+"gfs_phys",     "cldfra",       "cldfra",        "fv3_history",    "all",  .false.,  "none",  2
```
Comment (WalterKolczynski-NOAA) [inline parm/parm_fv3diag/diag_table:25]: What will happen if we try to run GFDL MP with this new variable? Right now cycling still only works with GFDL.

File: ush/forecast_postdet.sh [hunk 2]
```diff
@@ -343,7 +343,8 @@ EOF
   FNTSFC=${FNTSFC:-"$FIX_AM/RTGSST.1982.2012.monthly.clim.grb"}
   FNSNOC=${FNSNOC:-"$FIX_AM/global_snoclim.1.875.grb"}
   FNZORC=${FNZORC:-"igbp"}
-  FNAISC=${FNAISC:-"$FIX_AM/CFSR.SEAICE.1982.2012.monthly.clim.grb"}
+  #FNAISC=${FNAISC:-"$FIX_AM/CFSR.SEAICE.1982.2012.monthly.clim.grb"}
+  FNAISC=${FNAISC:-"$FIX_AM/IMS-NIC.blended.ice.monthly.clim.grb"}
```
Comment (WalterKolczynski-NOAA) [inline ush/forecast_postdet.sh:346]: Please remove the old line that has been commented out.

**Review by WalterKolczynski-NOAA — CHANGES_REQUESTED** [review]

**Comment by WalterKolczynski-NOAA** [comment]
In the future, this should probably be three separate PRs, even if they are one line (like the ice climo). PRs like that be handled very quickly without being held up by other changes.

You also haven't followed the PR template at all. Please update the first post of the PR to explain why these changes are being made. Listing the files modified is unnecessary; we can see by looking at the PR or the git log. End with a `Closes #XXX` to link the appropriate issue(s). Fill out the rest of the template sections on the type of change, how it was tested, and the checklist at the end. Also please shorten the PR title if possible (inability to formulate a short informative title is a clue a PR is doing too many different things).

File: ush/forecast_postdet.sh (hunk 2 above)
Comment (ChunxiZhang-NOAA) [reply ush/forecast_postdet.sh:346]: Done as suggested.

File: parm/parm_fv3diag/diag_table (hunk 1 above)
Comment (yangfanglin) [reply parm/parm_fv3diag/diag_table:25]: > What will happen if we try to run GFDL MP with this new variable? Right now cycling still only works with GFDL.

@WalterKolczynski-NOAA This should have no impact on cycling with GFDL MP, but I have not tested.

**Comment by JessicaMeixner-NOAA** [comment]
@yangfanglin @ChunxiZhang-NOAA - Is this PR waiting on a UFS_UTILS update (https://github.com/ufs-community/UFS_UTILS/pull/664) or is this PR ready to go as is?

**Comment by ChunxiZhang-NOAA** [comment]
@JessicaMeixner-NOAA  No, this PR is not waiting on the UFS_UTILS update. I am currently testing if all the runs, e.g., atmosphere only, S2SW, S2SWA, etc., can run successfully with the modifications in this PR.  After the tests, this PR is ready.

File: parm/parm_fv3diag/diag_table (hunk 1 above)
Comment (ChunxiZhang-NOAA) [reply parm/parm_fv3diag/diag_table:25]: I did an ATM only test with FV3_GFS_v16 suite and imp_physics=11 (GFDL). The test was successful.

**Review by WalterKolczynski-NOAA — APPROVED** [review]
Will merge tonight

**Comment by ChunxiZhang-NOAA** [comment]
The following tests passed: ATM only at C96, ATM only at C96 with the replacement of FV3_GFS_v16 suite and imp_physics=11, S2SW at C384 and S2SWA at C384.

**Comment by ChunxiZhang-NOAA** [comment]
> Will merge tonight

Great to know. Thank you Walter.

File: parm/parm_fv3diag/diag_table (hunk 1 above)
Comment (aerorahul) [reply parm/parm_fv3diag/diag_table:25]: This particular change breaks cycling (hybrid EnVar) when trying to compute ensemble mean from the surface files.
The `sfcfHHH.nc` file now contains a 4D variable `cldfra`.  I am not sure if it is a consequence of this line or elsewhere.  I apologize if it was introduced elsewhere.  

Screen Shot 2022-07-16 at 10 53 01 AM

As a result, a change in needed in the GSI utility computing surface ensemble mean.  Some basic debugging revealed Screen Shot 2022-07-16 at 10 52 32 AM

A test is running with the change in the GSI utility to skip this variable.
The test branch of GSI-utils is [here](https://github.com/NOAA-EMC/GSI-utils/compare/bugfix/getsfcensmeanp_cldfra?expand=1)

The test is being performed as part of PR #859.

Q. for possibly @yangfanglin and @RussTreadon-NOAA.
Should the variable be skipped or should the variable be averaged? 
The test above is skipping it (easy). 
To compute the average with this code will need work.

FYI. @WalterKolczynski-NOAA

Comment (RussTreadon-NOAA) [reply parm/parm_fv3diag/diag_table:25]: Add @emilyhcliu since the question at hand pertains to cloud fraction, `cldfra`.

Comment (yangfanglin) [reply parm/parm_fv3diag/diag_table:25]: > Q. for possibly @yangfanglin and @RussTreadon-NOAA. Should the variable be skipped or should the variable be averaged? The test above is skipping it (easy). To compute the average with this code will need work.
> 

@aerorahul   I think cldfra can be skipped for computing ensemble mean surface file.  @emilyhcliu if a cycled experiment is running with GFDL MP, cld_amt is the actual cloud fraction and is saved in atmf .  If running with Thompson MP, cldfra is the cloud fraction and saved in sfcf.   However, my impression is that even cld_amt was not used in any part of the DA in GFS.v16.

Comment (RussTreadon-NOAA) [reply parm/parm_fv3diag/diag_table:25]: Question for @yangfanglin .  Why does Thompson MP write cldfra to sfcfXXX?   It seems the 3d cloud fraction field should go in atmfXXX.  A 3d field is not a surface field.

Comment (aerorahul) [reply parm/parm_fv3diag/diag_table:25]: @RussTreadon-NOAA 
I was about to ask the same question.
`cldfra` is neither a "surface" field, nor a `2D` field.

Comment (yangfanglin) [reply parm/parm_fv3diag/diag_table:25]: > @RussTreadon-NOAA I was about to ask the same question. `cldfra` is neither a "surface" field, nor a `2D` field.

@aerorahul @RussTreadon-NOAA The model has two groups of outputs. dyn/atmf and phy/sfc.  For GFDL MP, cld_amt is a "prognostic" variable in the dycore group;  For Thompson MP, cldfra is a diagnostic variable from CCPP physics in the phy/sfc output group.  Placing variables across groups is technically difficult, if not impossible.  This is the major reason the 3d cldfra is written in sfc group.  I had long discussion with Jun (@junwang-noaa ) about this, but this is the best we can do given the current model infrastructure.    Please note @ChunxiZhang-NOAA is going to make a minor change to the diag_table to have cldfra labeled as fv3_history2d

**Pull request merged**